### PR TITLE
Hardening JH

### PR DESCRIPTION
**Pull request by henryju**

### CODE REVIEW

**Comment by henryju** [comment]
@bartfastiel I let you review the commits about SONAR-8727

File: server/sonar-server/src/main/java/org/sonar/server/computation/task/projectanalysis/issue/IssueAssigner.java [hunk 1]
```diff
@@ -75,11 +76,15 @@ public void onIssue(Component component, DefaultIssue issue) {
       String scmAuthor = guessScmAuthor(issue);
 
       if (!Strings.isNullOrEmpty(scmAuthor)) {
-        issueUpdater.setNewAuthor(issue, scmAuthor, changeContext);
+        if (scmAuthor.length() <= IssueDto.AUTHOR_MAX_SIZE) {
```
Comment (SonarTech) [inline server/sonar-server/src/main/java/org/sonar/server/computation/task/projectanalysis/issue/IssueAssigner.java]: MAJOR A "NullPointerException" could be thrown; "scmAuthor" is nullable here. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS2259)

**Comment by SonarTech** [comment]
SonarQube analysis reported 1 issue
* ![MAJOR][MAJOR] 1 major

Watch the comments in this conversation to review them.

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

**Comment by henryju** [comment]
@bartfastiel When you say "I created a user with a 288 character name" you mean "a 288 characters scm account" right?

**Comment by bartfastiel** [comment]
@henryju yes, exactly

**Comment by henryju** [comment]
@bartfastiel Strange because this is exactly what this IT is doing: https://github.com/SonarSource/sonarqube/pull/2705/files#diff-5ac53cb33fbdbaa88c627bf907899e6aR112

**Comment by bartfastiel** [comment]
I had forgotten, that the I have to set the git author email address, not the git author name.

File: tests/src/test/java/org/sonarqube/tests/issue/AutoAssignTest.java [hunk 2]
```diff
@@ -151,7 +151,12 @@ public void update_author_and_assignee_when_scm_is_activated() {
 
     // Authors and assignees are set
     for (Issue issue : issues) {
-      assertThat(issue.author()).isNotEmpty();
+      if (issue.line() == 10) {
+        // SONAR-8727
+        assertThat(issue.author()).isEmpty();
+      } else {
+        assertThat(issue.author()).isNotEmpty();
+      }
```
Comment (bartfastiel) [inline tests/src/test/java/org/sonarqube/tests/issue/AutoAssignTest.java]: I would prefer to make the test even more explicit, by using:
```
    assertThat(issues)
      .extracting(Issue::line, Issue::author)
      .containsExactlyInAnyOrder(
        tuple(1, "user1"),
        tuple(2, "user2"),
        tuple(3, "user3name"),
        tuple(4, "user4name"),
        tuple(5, "user5@email.com"),
        tuple(6, "user6@email.com"),
        tuple(7, "user7scmaccount"),
        tuple(8, "user8scmaccount"),
        tuple(9, "user8scmaccount"),
        tuple(10, "")
      );

```

(and on line 142:

```
    assertThat(issues)
      .extracting(Issue::line, Issue::author)
      .containsExactlyInAnyOrder(
        tuple(1, ""),
        tuple(2, ""),
        tuple(3, ""),
        tuple(4, ""),
        tuple(5, ""),
        tuple(6, ""),
        tuple(7, ""),
        tuple(8, ""),
        tuple(9, ""),
        tuple(10, "")
      );

```
)

**Pull request merged**